src/desktop/usb.rs

```diff
@@ -0,0 +1,259 @@
+                .get("ID_MODEL_FROM_DATABASE")
+                .or_else(|| properties.get("ID_MODEL_ENC"))
+                .and_then(|v| v.downcast_ref::<String>().ok())
+        })
+    }
+}
+
+/// Device to acquire.
+#[derive(SerializeDict, Type, Debug, Default)]
+#[zvariant(signature = "dict")]
+pub struct AcquireDevice {
+    writable: bool,
+}
+
+/// Option for AcquireDevice call.
+#[derive(SerializeDict, Type, Debug, Default)]
+#[zvariant(signature = "dict")]
+pub struct AcquireOptions {
+    handle_token: HandleToken,
+}
+
+/// Finished device acquired.
+#[derive(DeserializeDict, Type, Debug, Default)]
+#[zvariant(signature = "dict")]
+pub struct AcquiredDevice {
+    success: bool,
+    fd: Option<OwnedFd>,
+    error: Option<String>,
+}
+
+impl AcquiredDevice {
+    /// Return if the acquisition is a success.
+    pub fn success(&self) -> bool {
+        self.success
+    }
+
+    /// Return the error if it exists.
+    pub fn error(&self) -> Option<&str> {
+        self.error.as_deref()
+    }
+
+    /// Return the open file descriptor
+    pub fn fd(&self) -> Option<Fd> {
+        self.fd.as_ref().map(|fd| fd.into())
+    }
+}
+
+#[derive(Debug, Deserialize, Type)]
+/// A usb event received part of the `device_event` signal response.
+pub struct UsbEvent(/* action */ String, /* id */ String, UsbDevice);
+
+#[derive(Debug, Deserialize, Type)]
+/// A response received when the `device_event` signal is received.
+pub struct UsbDeviceEvent(OwnedObjectPath, Vec<UsbEvent>);
+
+impl UsbDeviceEvent {
+    /// The session triggered the state change
+    pub fn session_handle(&self) -> ObjectPath<'_> {
+        self.0.as_ref()
+    }
+
+    /// Events received
+    pub fn events(&self) -> &[UsbEvent] {
+        &self.1
+    }
+}
+
+impl UsbEvent {
+    /// The action.
+    pub fn event_action(&self) -> &str {
+        &self.0
+    }
+
+    /// The device id.
+    pub fn event_device_id(&self) -> &str {
+        &self.1
+    }
+
+    /// The [`UsbDevice`] properties.
+    pub fn event_device(&self) -> &UsbDevice {
+        &self.2
+    }
+}
+
+/// This interface provides access to USB devices.
+///
+#[derive(Debug)]
+#[doc(alias = "org.freedesktop.portal.Usb")]
+pub struct UsbProxy<'a>(Proxy<'a>);
+
+impl<'a> UsbProxy<'a> {
+    /// Create a new instance of [`UsbProxy`].
+    pub async fn new() -> Result<UsbProxy<'a>, Error> {
+        let proxy = Proxy::new_desktop("org.freedesktop.portal.Usb").await?;
+        Ok(Self(proxy))
+    }
+
+    /// Create Session
+    ///
+    #[doc(alias = "CreateSession")]
+    pub async fn create_session(&self) -> Result<Session<'a, Self>, Error> {
+        let options = CreateSessionOptions::default();
+        let session: OwnedObjectPath = self.0.call("CreateSession", &(&options)).await?;
+        Session::new(session).await
+    }
+
+    /// Enumerate USB devices.
+    ///
+    #[doc(alias = "EnumerateDevices")]
+    pub async fn enumerate_devices(
+        &self,
+        options: UsbOptions,
```

Comment:
There is no options here, so just pass Default? Also seems to have both a request_handle, yet it doesn't use the mechanism for requests and do a plain dbus call?